typescript.js

```diff
@@ -2,79 +2,60 @@ module.exports = {
-        '@typescript-eslint/no-array-constructor': 'error',
-        'no-dupe-class-members': 'off',
-        '@typescript-eslint/no-dupe-class-members': 'error',
-        'no-empty-function': 'off',
-        '@typescript-eslint/no-empty-function': ['error', {
-            allow: [
-                'arrowFunctions',
-                'functions',
-                'methods'
-            ]
-        }],
-        'no-extra-semi': 'off',
-        '@typescript-eslint/no-extra-semi': 'error',
-        'no-loop-func': 'off',
-        '@typescript-eslint/no-loop-func': 'error',
-        'no-redeclare': 'off',
-        '@typescript-eslint/no-redeclare': 'error',
-        'no-shadow': 'off',
-        '@typescript-eslint/no-shadow': 'error',
-        'no-unused-expressions': 'off',
-        '@typescript-eslint/no-unused-expressions': 'error',
-        'no-unused-vars': 'off',
-        '@typescript-eslint/no-unused-vars': ['error', { vars: 'all', args: 'after-used', ignoreRestSiblings: true }],
-        'no-use-before-define': 'off',
-        '@typescript-eslint/no-use-before-define': ['error', { functions: false, classes: true, variables: true }],
-        'no-useless-constructor': 'off',
-        '@typescript-eslint/no-useless-constructor': 'error',
-        quotes: 'off',
-        '@typescript-eslint/quotes': ['error', 'single', { avoidEscape: true }],
-        'no-return-await': 'off',
-        '@typescript-eslint/return-await': 'error',
-        semi: 'off',
-        '@typescript-eslint/semi': 'error',
-        'space-before-function-paren': 'off',
-        '@typescript-eslint/space-before-function-paren': ['error', {
-            anonymous: 'always',
-            named: 'never',
-            asyncArrow: 'always'
-        }],
-        'space-infix-ops': 'off',
-        '@typescript-eslint/space-infix-ops': 'error',
+
+        // None
+
+        /*
+            TypeScript rules outside of the recommended configuration
+        */
+        '@typescript-eslint/array-type': 'error',
+        '@typescript-eslint/ban-tslint-comment': 'error',
+        '@typescript-eslint/class-literal-property-style': 'error',
+        '@typescript-eslint/consistent-indexed-object-style': 'error',
+        '@typescript-eslint/consistent-type-assertions': 'error',
+        '@typescript-eslint/consistent-type-definitions': 'error',
+        '@typescript-eslint/consistent-type-imports': 'error',
+        '@typescript-eslint/explicit-function-return-type': 'error',
+        '@typescript-eslint/explicit-member-accessibility': 'error',
+        '@typescript-eslint/member-delimiter-style': 'error',
+        '@typescript-eslint/member-ordering': 'error',
+        '@typescript-eslint/method-signature-style': 'error',
+        '@typescript-eslint/no-confusing-non-null-assertion': 'error',
+        '@typescript-eslint/no-dynamic-delete': 'error',
+        '@typescript-eslint/no-extraneous-class': 'error',
+        '@typescript-eslint/no-implicit-any-catch': 'error',
```

Comment:
[@typescript-eslint/no-implicit-any-catch](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-implicit-any-catch.md)
Typescript doesn't support catching specific errors. With 4.0 it added support for catching `:any` or `:unknown`, but I don't see much benefit in catching `:unknown`, because you would have to use [Type Guards](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) or something similar to use the catched error. I suggest to disable this rule. We already enable the rule [no-throw-literal](https://eslint.org/docs/rules/no-throw-literal), which forces to only throw `Error` objects. So you can be sure that the catched object is at least an instance of `Error`, which is totally sufficient most of the time.

**Proposal**
`'@typescript-eslint/no-implicit-any-catch': 'off'`

Reply:
Shouldn't we be utilizing the type checker by defining the type, i.e. ```catch (e: Error)```? Are you more in favor of allowing implicit ```any```?

Reply:
I think that is not supported by TypeScript, otherwise I would be in favor of specifying the type to catch. 

![image](https://user-images.githubusercontent.com/9480438/107616498-7c160980-6c4e-11eb-921b-0a7f83325874.png)

Reply:
Thanks Lukas, you are right ([#20024](https://github.com/microsoft/TypeScript/issues/20024)).

Reply:
@rajsite this thread is awaiting your vote

Reply:
@LukasKall could you add a short comment to your proposal for no-implicit-any-catch explaining why we chose to turn it off (just to save Milan the effort of summarizing your comment)?